hbase/stackable/patches/phoenix/5.1.3/001-PHOENIX-omid-deps-version-bump.patch

```diff
@@ -0,0 +1,29 @@
+--- a/pom.xml	2024-05-14 13:20:33.185420698 +0200
++++ b/pom.xml	2024-05-15 11:16:29.036294238 +0200
+@@ -1304,7 +1304,7 @@
+       <dependency>
+         <groupId>org.apache.omid</groupId>
+         <artifactId>omid-transaction-client</artifactId>
+-        <version>${omid.version}</version>
++        <version>1.1.2</version>
+         <exclusions>
+           <exclusion>
+             <groupId>org.testng</groupId>
+@@ -1315,7 +1315,7 @@
+       <dependency>
+         <groupId>org.apache.omid</groupId>
+         <artifactId>omid-commit-table</artifactId>
+-        <version>${omid.version}</version>
++        <version>1.1.2</version>
+         <exclusions>
+           <exclusion>
+             <groupId>org.testng</groupId>
+@@ -1326,7 +1326,7 @@
+       <dependency>
+         <groupId>org.apache.omid</groupId>
+         <artifactId>omid-common</artifactId>
+-        <version>${omid.version}</version>
++        <version>1.1.2</version>
```

Comment:
Couldn't this just be a bump to `omid.version`, similarly to the JRuby patch above?

Reply:
Oh, yikes. I'd rather consider the held-back versions the exception, but that also raises followup questions like.. "is omid 1.1.2 actually compatible with omid-hbase-common-hbase2.x 1.1.1? why did they need to drop it here?".

(To be clear, I'm not complaining at you here, more just being cautious about what the implications would be...)

Reply:
Seems like in the next phoenix version (5.2.0) they replaced omid-hbase-common-hbase2.x with omid-hbase-common (I base this observation solely on the name), which seems to be compatible with omid 1.1.2, they updated these versions as well. https://repo1.maven.org/maven2/org/apache/phoenix/phoenix/5.2.0/phoenix-5.2.0.pom

So the vulnerability coming from phoenix should be fixed with a new phoenix version, no patch needed then?

Reply:
Honestly, I'd have to leave that question to someone more knowledgeable about Phoenix than me..

Reply:
After some more digging: phoenix versions 5.x require Hadoop3, which HBase 2.x does not built with out of the box. Since we are building from source we might change that but I'm not sure if that has some other implications or problems down the line (https://github.com/apache/phoenix/blob/master/BUILDING.md#hbase-2-and-hadoop-3)
So not sure if it would be more reasonable to update the phoenix version together with the HBase version.